pallets/grant/src/lib.rs

```diff
@@ -322,15 +330,14 @@ pub mod pallet {
 		// Function that allows funds to be sent to winner
 		fn transfer_funds_to_winner() -> Result<(), DispatchError> {
 
-			let (_lottery_account, lottery_balance) = Self::treasury_account();
+			let (treasury_account, treasury_balance) = Self::treasury_account();
+			let grant_total = T::GrantAmount::get();
 
-			let treasury= &Self::account_id();
-
-			// TODO: Implement formula that grants based on total supply and not whole balance
-			let _total = T::Currency::total_issuance();
+			ensure!(treasury_balance > grant_total, Error::<T>::TreasuryEmpty);
```

Comment:
Can we also implement check to ensure that only empty account balances can request grants? Or at least accounts with less than the Existential deposit?

Reply:
to become a requester account balance must be zero, after reading the glossary @ https://docs.substrate.io/reference/glossary/ i think the current implementation is actually wrong.

An account with balance of 0 is below the ED and there for is dropped from storage and reaped. Thus, it should be only account with balance of ED can apply for a grant.

Reply:
https://github.com/UniversalDot/universal-dot-node/pull/107/commits/97db732c50a051b81ecb819d27f0a858a67d2596